pkg/core/blockchain.go

```diff
@@ -1199,9 +1196,25 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
 	if t.ValidUntilBlock <= height || t.ValidUntilBlock > height+transaction.MaxValidUntilBlockIncrement {
 		return errors.Errorf("transaction has expired. ValidUntilBlock = %d, current height = %d", t.ValidUntilBlock, height)
 	}
+	balance := util.Fixed8FromInt64(bc.GetNEP5Balances(t.Sender).Trackers[bc.contracts.GAS.Hash].Balance)
+	totalSenderFee := util.Fixed8FromInt64(0)
+	for _, txWithFee := range bc.memPool.GetVerifiedTransactions() {
+		if txWithFee.Tx.Sender.Equals(t.Sender) {
+			totalSenderFee += txWithFee.Tx.NetworkFee + txWithFee.Tx.SystemFee
+		}
+	}
+	fee := t.SystemFee + t.NetworkFee + totalSenderFee
+	if balance < fee {
+		return errors.New("insufficient funds")
+	}
 	if io.GetVarSize(t) > transaction.MaxTransactionSize {
 		return errors.Errorf("invalid transaction size = %d. It shoud be less then MaxTransactionSize = %d", io.GetVarSize(t), transaction.MaxTransactionSize)
 	}
+	// TODO
+	//net_fee := t.NetworkFee - io.GetVarSize(t) * native.Contracts.Policy.GetFeePerByte(bc);
```

Comment:
You can reuse `(*Blockchain).FeePerByte` for now, maybe making it more like `native.Contracts.Policy.GetFeePerByte`.

Reply:
Does the following code suite here or we have to use storage?
```
func (bc *Blockchain) FeePerByte(*transaction.Transaction) util.Fixed8 {
	return util.Fixed8FromInt64(1000)
}
```
(see https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/Native/PolicyContract.cs#L61)

Reply:
It's OK hardcoding it for now, we'll fix it in #904.